### PR TITLE
Create documents required for project guidelines

### PR DESCRIPTION
**Pull request by jhsi517**
Added the following documents to provide project guidelines:

- Contribution Guideline
- Code of Conduct
- React JS Conventions and Guideline
- C# Conventions and Guideline

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+It is important for us to keep the core code clean and consistent. Please ensure any pull request have been reviewed by at least one of the team members.
+
+### Steps for pull request
+
+Above steps should be used when following the steps belowwhen you'd like your work considered for inclusion in the
+project:
+
+1. [Fork](http://help.github.com/fork-a-repo/) the project, clone your fork,
+   and configure the remotes:
```
Comment (JedLJRobertson) [inline CONTRIBUTING.md]: Do we need people to fork, or do we just clone and use feature branches? I think the assignment brief suggests the latter.

File: C#-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# C# Coding Guidelines
```
Comment (JedLJRobertson) [inline C#-guidelines.md]: This looks good, but are these written or sourced from somewhere?

If so, we should have a citation for where these came from.

File: CONTRIBUTING.md (hunk 1 above)
Comment (JedLJRobertson) [reply CONTRIBUTING.md]: NVM, we do need to fork

File: C#-guidelines.md (hunk 2 above)
Comment (jhsi517) [reply C#-guidelines.md]: They are sourced and adjusted. Have included the references in the last section :)

**Comment by jhsi517** [comment]
> Just a FYI, the assignment says that all commits should be squashed before merging.

Sorry, thought it was optional for some reason. I have just squashed the commits. Please review and confirm if that's alright :)

**Comment by jhsi517** [comment]
> Have any of these been written by us or are they entirely copy pasted and then referenced? The assignment brief states that we need contributing guidelines, code of conduct etc... but I would have thought the expectation would be that we wrote these ourselves

Generally, I have extract parts of different documents from different places and made a few minor changes to the content. The sources have been referenced at the bottom. Have raised this question to Kelly last week. Please see below for her response: 

Hi Lucas,
For the Code of Conduct, there are actually boilerplates that are publicly available, like the Contributor Covenant. I suggest you use something like this and modify as needed to suit your project with correct attribution. I’ll cover this in the lecture on Monday. 

For the contribution guidelines, you can certainly be inspired by other contribution guidelines. Consider the guidance on what should be included in the project descriptor and pay attention to licenses of the projects you might borrow material from. Attribution may be required, but also you may need to use the same license as that project if you copy parts of their contribution guidelines.

**Comment by HongShi10** [comment]
I think what Lucas has now is fine and should be a template that we can base off for now. If we need adjusting then we can just change it if need.

**Comment by jhsi517** [comment]
Closing this pull request since it did not meet the branch naming convention agreed by the team. A new pull request has been raised.

**Pull request closed without merging**